OpenRA.Mods.Common/Activities/Move/Move.cs

```diff
@@ -326,13 +325,15 @@ public MovePart(Move move, WPos from, WPos to, int fromFacing, int toFacing, int
 
 			public override bool Cancel(Actor self, bool keepQueue = false)
 			{
-				Move.Cancel(self, keepQueue);
-				return base.Cancel(self);
+				// MovePart has no child or queued activity so it is correct to cancel this
+				// then return the RootActivity's cancel result.
+				base.Cancel(self);
+				return RootActivity.Cancel(self, keepQueue);
 			}
 
 			public override void Queue(Activity activity)
 			{
-				Move.Queue(activity);
+				ParentActivity.Queue(activity);
```

Comment:
see comment above wrt to the `return ChildActivity` line.

Btw, the `readonly Move move` field can either be removed, or be assigend to `ParentActivity as Move` in the ctor, and then be used instead of ParentActivity. You'd avoid all the casts below that way.